### PR TITLE
Add common spec and run validation against it

### LINKED ISSUE #19: [META 27] Validate logger against the spec
Sync the spec to the repo and validate the logger against the spec, similar to what the java logger is doing in https://github.com/elastic/ecs-logging/issues/30.

### PR DESCRIPTION
**Pull request by simitt**
This PR copies the common [spec](https://github.com/elastic/ecs-logging/blob/master/spec/spec.json) for ECS loggers and runs validations for the fields that are already supported. 

The spec itself is just manually copied over, as according to https://github.com/elastic/observability-robots/issues/313 the spec should automatically be synced when changed. 

While on it, a bug was found in the `timestamp` format, where the timestamp can be changed from the default `ISO8601` format to any other format. With this fix the timestamp will always be `ISO8601` for the ECS formatter. 

closes #19

### CODE REVIEW

File: logger_test.go [hunk 1]
```diff
@@ -42,17 +46,65 @@ func (tw *testOutput) Write(p []byte) (int, error) {
 
 func (tw *testOutput) Sync() error { return nil }
 
-func (tw *testOutput) requireContains(t *testing.T, keys []string) {
+func (tw *testOutput) validate(t *testing.T, keys []string) {
```
Comment (axw) [inline logger_test.go:49]:
```suggestion
func (tw *testOutput) validate(t *testing.T, keys ...string) {
```

nit

File: logger_test.go [hunk 2]
```diff
@@ -42,17 +46,65 @@ func (tw *testOutput) Write(p []byte) (int, error) {
...
+	// skip Sanitization as it is not implemented yet
+	for name, val := range tw.m {
+		field, ok := spec.V1.Fields[name]
+		if !ok { // custom field not defined in spec
+			continue
+		}
+		if field.Required { // all required fields must be present in the log line
+			require.Contains(t, tw.m, name)
+			require.NotNil(t, tw.m[name])
+		}
```
Comment (axw) [inline logger_test.go:75]: This will never fail, because we're ranging over `tw.m` rather than `spec.V1.Fields`. We should swap that around.

**Review by axw — CHANGES_REQUESTED** [review]
Mostly LGTM, just a couple of small things

File: logger_test.go (hunk 2 above)
Comment (simitt) [reply logger_test.go]: good catch - updated

**Review by axw — APPROVED** [review]

**Pull request merged**